Dockerfile

```diff
@@ -0,0 +1,62 @@
+FROM ubuntu:24.04
+
+# Install dependencies
+RUN apt update && apt install -y \
+    g++ \
+    gcc \
+    m4  \
+    make \
+    patch \
+    texinfo \
+    wget \
+    xz-utils
+
+# Build iconv
```

Comment:
Since future versions of gdb, iconv, gmp and mpfr may change in the future, i prefer you placing everything here onwards into a separate `build.sh` file, and then run that file using `docker run <docker name> .... build.sh`

That way, the dockerfile will not needed to be rebuilt in every time a single dependency is changed, and thus waste tons of space due to old images adding up (yes, one can use `docker image prune` to clean them up, but if we can avoid that, we should :) ).

(It is also worth noting that docker will cache every step throughout the way, which will may make the dockerfile unnecessary huge)